classes/plugins/PubObjectsExportPlugin.inc.php

```diff
@@ -163,8 +163,13 @@ function executeExportAction($request, $objects, $filter, $tab, $objectsFileName
 			assert($filter != null);
 			// Get the XML
 			$exportXml = $this->exportXML($objects, $filter, $context);
-			header('Content-type: application/xml');
-			echo $exportXml;
+			$exportFileName = $this->getExportFileName($objectsFileNamePart, $context);
+			file_put_contents($exportFileName, $exportXml);
```

Comment:
Some of this is starting to reinvent the TemporaryFileManager; I haven't worked with that class for a while, but I think it should be pretty quick to make use of it instead for management of file creation/deletion/etc.